examples/c/ex_tiered.c

```diff
@@ -0,0 +1,188 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * ex_tiered.c
+ *	This is an example demonstrating how to create and connect to a
+ *	database running tiered storage.
+ */
+#include <test_util.h>
+
+/*
+ * Open the uri and starting at the first indicated key, insert count entries.
+ */
+static void
+add_data(WT_SESSION *session, const char *uri, int first_key, int count)
+{
+    WT_CURSOR *cursor;
+    int i;
+
+    error_check(session->open_cursor(session, uri, NULL, NULL, &cursor));
+
+    /* Insert a set of key/data pairs. */
+    for (i = first_key; i < first_key + count; i++) {
+        cursor->set_key(cursor, i);
+        cursor->set_value(cursor, "value");
+        error_check(cursor->insert(cursor));
+    }
+
+    error_check(cursor->close(cursor));
+}
+
+/*
+ * Show all entries found in the uri.
+ */
+static void
+show_data(WT_SESSION *session, const char *uri, const char *comment)
+{
+    WT_CURSOR *cursor;
+    int key, ret;
+    const char *value;
+
+    printf("%s:  %s\n", uri, comment);
+
+    error_check(session->open_cursor(session, uri, NULL, NULL, &cursor));
+
+    while ((ret = cursor->next(cursor)) == 0) {
+        cursor->get_key(cursor, &key);
+        cursor->get_value(cursor, &value);
+        printf(" %d: %s\n", key, value);
+    }
+
+    /* We expect "not found" to end the iteration.  Anything else is an error. */
```

Comment:
This may or may not be true as WT_ROLLBACK can be returned if we revert the changes in WT-10133. I agree with the code though.

Reply:
I don't think we should change this "not found" comment until or unless WT-10133 is reverted.  If we do revert WT-10133 (I'm opposed to this!) we'll need to change a lot of comments and a lot of code in lots of examples.